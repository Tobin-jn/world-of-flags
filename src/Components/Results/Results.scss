.Results {
  background: rgba(240, 240, 240, .99);
  border-radius: 8px;
  box-shadow: 0 0 1rem rgba(0, 0, 0, .3);
  display: inline-flex;
  flex-direction: column;
  height: 55%;
  justify-content: space-between;
  left: 2%;
  padding: 5rem 0;
  position: absolute;
  text-align: center;
  top: 8rem;
  width: 96%;
  z-index: 2;
}

p {
  font-size: 2rem;
  margin: 0;
}

.results-title {
  font-size: 3rem;
}

.points-gained {
  font-size: 2.5rem;
  // margin-bottom: 2rem;
}

.results-divider {
  background-color: #c2c2c2;
  height: 1px;
  margin: 0 auto;
  width: 80%;
}

.total-points-count {
  font-size: 3rem;
  font-weight: 700;
  // margin-bottom: 1rem;
}