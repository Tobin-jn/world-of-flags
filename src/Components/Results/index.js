import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import './Results.scss';

class Results extends Component {
  constructor(props) {
    super(props);
  }

  handleClick = () => {
    const { getCountry, closeResults } = this.props;

    getCountry()
    closeResults();
  }

  render() {
    const { status, correctCountry, points, totalPoints } = this.props;
    return (
      <div className='Results'>
        <p className='results-title'>{ status }!</p>
        <p className='results-correct-country'>{ correctCountry } was the correct answer</p>
        <p className='points-gained'><strong>+{ points }</strong> pts</p>
        <div className='results-divider'></div>
        <p>Total</p>
        <p className='total-points-count'>{ totalPoints }</p>
        <div className='button results-continue' onClick={this.handleClick}>Continue</div>
      </div>
    );
  }
}

export default Results;
