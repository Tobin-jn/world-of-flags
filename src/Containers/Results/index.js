import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import './Results.scss';

class Results extends Component {
  render() {
    return (
      <div className='Results'>
        <div className='account-area'>
          <h5 className='account-text'>Alex <strong>25</strong></h5>
        </div>
      </div>
    );
  }
}

export default Results;
