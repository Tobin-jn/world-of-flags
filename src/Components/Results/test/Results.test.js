import React from 'react';
import { shallow, mount } from 'enzyme';
import Results from '../index';

describe('Results', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Results />)
  });
  
    it('should render like the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  it('should post if the guess was correct', () => {
  });

  it('should post if the guess was incorrect', () => {
  });

  it('should show the amount of points gained from last session', () => {
  });

  it('should show the cumulative amount of points', () => {
  })

});