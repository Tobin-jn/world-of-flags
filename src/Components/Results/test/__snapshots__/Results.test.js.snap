// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results handleClick should render like the snapshot 1`] = `undefined`;

exports[`Results should render like the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Results />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="results-title"
        >
          !
        </p>,
        <p
          className="results-correct-country"
        >
           was the correct answer
        </p>,
        <p
          className="points-gained"
        >
          <strong>
            +
          </strong>
           pts
        </p>,
        <div
          className="results-divider"
        />,
        <p>
          Total
        </p>,
        <p
          className="total-points-count"
        />,
        <div
          className="button results-continue"
          onClick={[Function]}
        >
          Continue
        </div>,
      ],
      "className": "Results",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            "!",
          ],
          "className": "results-title",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          "!",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " was the correct answer",
          ],
          "className": "results-correct-country",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " was the correct answer",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              +
            </strong>,
            " pts",
          ],
          "className": "points-gained",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "+",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "+",
              undefined,
            ],
            "type": "strong",
          },
          " pts",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "results-divider",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Total",
        },
        "ref": null,
        "rendered": "Total",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "total-points-count",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Continue",
          "className": "button results-continue",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Continue",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="results-title"
          >
            !
          </p>,
          <p
            className="results-correct-country"
          >
             was the correct answer
          </p>,
          <p
            className="points-gained"
          >
            <strong>
              +
            </strong>
             pts
          </p>,
          <div
            className="results-divider"
          />,
          <p>
            Total
          </p>,
          <p
            className="total-points-count"
          />,
          <div
            className="button results-continue"
            onClick={[Function]}
          >
            Continue
          </div>,
        ],
        "className": "Results",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              "!",
            ],
            "className": "results-title",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            "!",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " was the correct answer",
            ],
            "className": "results-correct-country",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " was the correct answer",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                +
              </strong>,
              " pts",
            ],
            "className": "points-gained",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "+",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "+",
                undefined,
              ],
              "type": "strong",
            },
            " pts",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "results-divider",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Total",
          },
          "ref": null,
          "rendered": "Total",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "total-points-count",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Continue",
            "className": "button results-continue",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Continue",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
